deepchecks/nlp/utils/text_properties.py

```diff
@@ -335,6 +338,57 @@ def unique_noun_count(raw_text: Sequence[str]) -> List[str]:
     return result
 
 
+def automated_readability_index(raw_text: Sequence[str]) -> List[str]:
+    """Return a list of floats of Automated Readability Index (ARI) per text sample.
+
+    The Automated Readability Index (ARI) is a formula that uses the number of characters,
+    words, and sentences in a text to calculate a score that represents the grade level required
+    to understand the text. The resulting score is a whole number that represents a U.S. grade level.
+    For more information: https://en.wikipedia.org/wiki/Automated_readability_index
+    """
+    if not nltk_download('punkt', quiet=True):
+        warnings.warn('nltk punkt not found, automated readability index cannot be calculated.'
+                      ' Please check your internet connection.', UserWarning)
+        return [np.nan] * len(raw_text)
+    result = []
+    for text in raw_text:
+        if not pd.isna(text):
+            sentence_count = len(sent_tokenize(text))
+            word_count = len(re.findall(r'\w+', text))
+            char_count = len(re.sub(r'\s', '', text))
+            try:
```

Comment:
try catch is slow, can replace it with:
if word_count ==0 or sentence_count==0:
result.append(1) # smallest ari index
else:
...

Reply:
Done